Jack/add doc gen docs (#43)
anz-bank/sysl-website: blog/2020-07-23-sysl-catalog.md

A couple of months pass and now there’s another team which relies on team A’s service heavily. They are releasing soon and need to create release documentation; so they decide to use the same method that Team A is using.

Now there’s a problem; there are two teams with two separate sets of documentation. Some of it is manual and some of it is automated. This can cause problems for multiple reasons;

Manually written parts might need to be repeated and might fall out of sync across the two sources
If manually written documentation isn't repeated then the representation of their dependency is limited by a hyperlink to Team A's documentation without fully integrating
The decoupling of documentation and code means that the documentation is most likely going to fall out of date
Because there’s no persistence between team A’s documentation and Team B; if team A make a very specific change it likely won’t show in team B’s documentation
This is what sysl-catalog is trying to solve

sysl-catalog uses the sysl language as an intermediary between different formats to be able to generate different views of how services work

## What is sysl?

> Sysl is a “system specification language”; think of it like swagger or protos, but a much higher level, and with the ability to represent not only types, applications and endpoints, but interactions between those applications and endpoints; it plans to define what the code does itself.

Reviewer: ```suggestion
Sysl is a “system specification language”. It is similar to other specification languages like Swagger that model types and endpoints, but it also captures interactions between those endpoints. This provides a richer set of information from which to generate outputs like documentation and code.
```

## What does sysl-catalog do?

Sysl-catalog is just a static site generator.

> Sysl-catalog parses a sysl file (with the .sysl extension) and represents it in a visual form; It can represent endpoints (in sequence diagrams) request/response types or database tables, as well as integration diagrams.

Reviewer: ```suggestion
Sysl-catalog parses a Sysl file (with the `.sysl` extension) and represents it in a visual form. It can represent endpoints (in sequence diagrams) request/response types or database tables, as well as integration diagrams.
```

> It uses go's `text/template` to do this, and if any addition is needed to be made, custom templates can be used (see templates for examples)

Reviewer: ```suggestion
It uses Go's `text/template` to do this, and custom templates can be used to make changes (see templates for examples)
```
